### ISSUE TITLE
Font atlas rendering issues (texture cache?) [Cat Cafe Manager 01009BE014CA6000]

### ISSUE THREAD

**Issue by lostromb** [Ryujinx/Ryujinx#4225]
### What's the issue you encountered?

In the initial loading screen for Cat Cafe Manager (and in a few subsequent places in regular gameplay), it appears that the font atlas gets corrupted somehow, which causes a few random glyphs in a sentence to not appear.

It's semi-random when it happens, but there's a very distinct moment when it does. Text will be rendering normally, and then on the very next frame that same text will be corrupted, and it will stay that way for the rest of the loading sequence. Repros on OpenGL and Vulkan.

Renderdoc output. Vertex inputs are correct, but the texcoords refer to currently blank areas in the font atlas (text should say "Bundling tall tales...")
![image](https://user-images.githubusercontent.com/918053/211165821-593cf8bd-504e-4fb3-a992-6463845ae7a3.png)

The font atlas. It's a 512x512 texture with glyphs extending to about the first 200 pixels across. This also appears to be the wrong atlas as it only contains the glyphs needed for "Counting kittens..."
![image](https://user-images.githubusercontent.com/918053/211166050-2f449c86-42aa-465d-bd30-93fc0a3de2b8.png)

I can see individual writes to the regions of the subtexture where the sampler expects glyphs to be, but for some reason the input data for these copies was blank, or the texture cache is misbehaving, or this is the wrong atlas texture somehow
![image](https://user-images.githubusercontent.com/918053/211166066-ec0b1530-9b55-45bf-8bb1-46ebc4a82245.png)

### How can the issue be reproduced?

Start the game and watch the load screen.

### Log file

[Ryujinx_1.1.518_2023-01-07_18-46-08.log](https://github.com/Ryujinx/Ryujinx/files/10367019/Ryujinx_1.1.518_2023-01-07_18-46-08.log)

### Environment?

- Ryujinx version: 1.1.518</br>
- Game version: 1.0.399</br>
- System Specs:
  - OS: Windows 11
  - CPU: Ryzen 3700X
  - GPU: NVidia 1650 Super
  - RAM: 64Gb
- Applied Mods : No

**Comment by gdkchan** [comment]
Does it happen with backend threading set to Off?

**Comment by lostromb** [comment]
> Does it happen with backend threading set to Off?

Yes.

![image](https://user-images.githubusercontent.com/918053/211167001-0fda554f-1f25-4a85-95ba-85df7be9ad16.png)

**Comment by gdkchan** [comment]
It seems similar to #3641. Someone said that setting the backend multithreading to off fixed the issue on one of those games IIRC, that's why I thought it would be interesting to try here. It is hard to say what the real problem is, OpenGL games does a lot of weird things with textures, including copying their data in chuncks, reading from CPU and copying it on the CPU side. Anything wrong in this process will cause the texture to be broken later on. It would be even harder to debug if the textures are not consistently broken.

**Comment by lostromb** [comment]
Yeah, I looked at a few other instances of font rendering issues (in Diablo 2 and various visual novels) and got renderdoc traces of each. While those other games appear to have a bug in software rendering (that is, they render the glyphs on CPU and then blit the entire rendered sentence to the output buffer directly), this case seemed unique because I can see it building the atlas texture manually and then rendering each individual character to triangles using `drawBuffers`, and then somewhere along the way Ryujinx is messing up the texture binding or texture cache somehow.
